tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -434,20 +437,57 @@ func (iptConfigurator *IptablesConfigurator) run() {
 	iptConfigurator.handleInboundIpv4Rules(ipv4RangesInclude)
 	iptConfigurator.handleInboundIpv6Rules(ipv6RangesExclude, ipv6RangesInclude)
 
-	// Execute iptables commands
-	for _, cmd := range iptConfigurator.iptables.BuildV4() {
-		if len(cmd) > 1 {
-			iptConfigurator.ext.RunOrFail(cmd[0], cmd[1:]...)
-		} else {
-			iptConfigurator.ext.RunOrFail(cmd[0])
-		}
+	iptConfigurator.executeCommands()
+}
+
+func (iptConfigurator *IptablesConfigurator) createRulesFile(filePath string, contents string) {
+	f, err := os.Create(filePath)
+	if err != nil {
+		fmt.Println("Unable to create file for iptables-restore")
```

Comment:
Return an `error` instead of writing into the standard output and exiting the process.
That will make this function reusable and unit-testable.